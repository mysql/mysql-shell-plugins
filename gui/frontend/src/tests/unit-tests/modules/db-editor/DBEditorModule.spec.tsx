/*
 * Copyright (c) 2022, 2024, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is designed to work with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have either included with
 * the program or referenced in the documentation.
 *
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import icon from "../../../../assets/images/modules/module-sql.svg";

import { createRef } from "preact";
import { mount } from "enzyme";

import { DBEditorModuleId } from "../../../../modules/ModuleInfo.js";
import { DBEditorModule } from "../../../../modules/db-editor/DBEditorModule.js";

describe("DBEditor module tests", (): void => {

    it("Test DBEditorModule instantiation", () => {
        const innerRef = createRef<HTMLButtonElement>();
        const component = mount<DBEditorModule>(
            <DBEditorModule
                innerRef={innerRef}
            />,
        );
        const props = component.props();
        expect(props.innerRef).toEqual(innerRef);
        expect(DBEditorModule.info).toStrictEqual({
            id: DBEditorModuleId,
            caption: "DB Editor",
            icon,
        });
        expect(component).toMatchSnapshot();
        component.unmount();
    });

});
