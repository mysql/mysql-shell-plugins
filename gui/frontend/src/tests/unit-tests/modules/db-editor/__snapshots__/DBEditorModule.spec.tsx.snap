// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DBEditor module tests Test DBEditorModule instantiation 1`] = `
<DBEditorModule
  innerRef={
    {
      "current": <div
        class="msg container fixedScrollbar tabview top dbModuleTabview moduleHost"
        style="flex-direction: column; --tabItem-border-width: 0px; --content-separator-width: 0px;"
      >
        <div
          class="msg container tabAreaContainer"
          style="flex-direction: row;"
        >
          <div
            class="scrollable"
          >
            <div
              class="msg container tabArea"
              style="flex-direction: row;"
            >
              <div
                class="msg button tabItem selected"
                draggable="true"
                id="progress"
                role="button"
                tabindex="-1"
              >
                <span
                  class="msg icon"
                  data-tooltip="inherit"
                  style="mask-image: url(overviewPage.svg);"
                />
                <label
                  class="msg label"
                  data-tooltip="inherit"
                >
                  Open Connection
                </label>
              </div>
            </div>
            <div
              class="scrollbar"
            >
              <div
                class="slider"
              />
            </div>
          </div>
        </div>
        <div
          class="msg container fixedScrollbar content"
          style="flex-direction: column;"
        >
          <div
            class="msg container fixedScrollbar progressIndicatorHost"
            id="loadingProgressIndicator"
            style="flex-direction: column; justify-content: center; align-items: center; opacity: 0.95; --position: 0; --width: 80; --height: 80; --strokeWidth: 5;"
          >
            <svg
              class="circleBackground animated"
            >
              <circle
                cx="37.5"
                cy="37.5"
                r="35"
              />
              <circle
                cx="37.5"
                cy="37.5"
                r="35"
              />
            </svg>
          </div>
          <label
            class="msg label"
            data-tooltip="expand"
            id="progressMessageId"
          />
        </div>
      </div>,
    }
  }
>
  <Tabview
    canReorderTabs={true}
    className="dbModuleTabview moduleHost"
    hideSingleTab={false}
    innerRef={
      {
        "current": <div
          class="msg container fixedScrollbar tabview top dbModuleTabview moduleHost"
          style="flex-direction: column; --tabItem-border-width: 0px; --content-separator-width: 0px;"
        >
          <div
            class="msg container tabAreaContainer"
            style="flex-direction: row;"
          >
            <div
              class="scrollable"
            >
              <div
                class="msg container tabArea"
                style="flex-direction: row;"
              >
                <div
                  class="msg button tabItem selected"
                  draggable="true"
                  id="progress"
                  role="button"
                  tabindex="-1"
                >
                  <span
                    class="msg icon"
                    data-tooltip="inherit"
                    style="mask-image: url(overviewPage.svg);"
                  />
                  <label
                    class="msg label"
                    data-tooltip="inherit"
                  >
                    Open Connection
                  </label>
                </div>
              </div>
              <div
                class="scrollbar"
              >
                <div
                  class="slider"
                />
              </div>
            </div>
          </div>
          <div
            class="msg container fixedScrollbar content"
            style="flex-direction: column;"
          >
            <div
              class="msg container fixedScrollbar progressIndicatorHost"
              id="loadingProgressIndicator"
              style="flex-direction: column; justify-content: center; align-items: center; opacity: 0.95; --position: 0; --width: 80; --height: 80; --strokeWidth: 5;"
            >
              <svg
                class="circleBackground animated"
              >
                <circle
                  cx="37.5"
                  cy="37.5"
                  r="35"
                />
                <circle
                  cx="37.5"
                  cy="37.5"
                  r="35"
                />
              </svg>
            </div>
            <label
              class="msg label"
              data-tooltip="expand"
              id="progressMessageId"
            />
          </div>
        </div>,
      }
    }
    onSelectTab={[Function]}
    pages={
      [
        {
          "caption": "Open Connection",
          "content": <p>
            <ProgressIndicator
              backgroundOpacity={0.95}
              id="loadingProgressIndicator"
              linear={false}
            />
            <Label
              caption=""
              id="progressMessageId"
            />
          </p>,
          "icon": "overviewPage.svg",
          "id": "progress",
        },
      ]
    }
    selectedId="progress"
    showTabs={true}
    stretchTabs={false}
    tabPosition="top"
  >
    <Container
      className="msg tabview top dbModuleTabview moduleHost"
      innerRef={
        {
          "current": <div
            class="msg container fixedScrollbar tabview top dbModuleTabview moduleHost"
            style="flex-direction: column; --tabItem-border-width: 0px; --content-separator-width: 0px;"
          >
            <div
              class="msg container tabAreaContainer"
              style="flex-direction: row;"
            >
              <div
                class="scrollable"
              >
                <div
                  class="msg container tabArea"
                  style="flex-direction: row;"
                >
                  <div
                    class="msg button tabItem selected"
                    draggable="true"
                    id="progress"
                    role="button"
                    tabindex="-1"
                  >
                    <span
                      class="msg icon"
                      data-tooltip="inherit"
                      style="mask-image: url(overviewPage.svg);"
                    />
                    <label
                      class="msg label"
                      data-tooltip="inherit"
                    >
                      Open Connection
                    </label>
                  </div>
                </div>
                <div
                  class="scrollbar"
                >
                  <div
                    class="slider"
                  />
                </div>
              </div>
            </div>
            <div
              class="msg container fixedScrollbar content"
              style="flex-direction: column;"
            >
              <div
                class="msg container fixedScrollbar progressIndicatorHost"
                id="loadingProgressIndicator"
                style="flex-direction: column; justify-content: center; align-items: center; opacity: 0.95; --position: 0; --width: 80; --height: 80; --strokeWidth: 5;"
              >
                <svg
                  class="circleBackground animated"
                >
                  <circle
                    cx="37.5"
                    cy="37.5"
                    r="35"
                  />
                  <circle
                    cx="37.5"
                    cy="37.5"
                    r="35"
                  />
                </svg>
              </div>
              <label
                class="msg label"
                data-tooltip="expand"
                id="progressMessageId"
              />
            </div>
          </div>,
        }
      }
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      orientation="column"
      style={
        {
          "--content-separator-width": "0px",
          "--tabItem-border-width": "0px",
        }
      }
    >
      <div
        className="msg container fixedScrollbar tabview top dbModuleTabview moduleHost"
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        style={
          {
            "--content-separator-width": "0px",
            "--tabItem-border-width": "0px",
            "alignItems": undefined,
            "flexDirection": "column",
            "flexWrap": undefined,
            "justifyContent": undefined,
          }
        }
      >
        <Container
          className="tabAreaContainer"
          fixedScrollbars={false}
          orientation="row"
        >
          <div
            className="msg container tabAreaContainer"
            style={
              {
                "alignItems": undefined,
                "flexDirection": "row",
                "flexWrap": undefined,
                "justifyContent": undefined,
              }
            }
          >
            <div
              className="scrollable"
              onWheel={[Function]}
            >
              <Container
                className="tabArea"
                fixedScrollbars={false}
                innerRef={
                  {
                    "current": <div
                      class="msg container tabArea"
                      style="flex-direction: row;"
                    >
                      <div
                        class="msg button tabItem selected"
                        draggable="true"
                        id="progress"
                        role="button"
                        tabindex="-1"
                      >
                        <span
                          class="msg icon"
                          data-tooltip="inherit"
                          style="mask-image: url(overviewPage.svg);"
                        />
                        <label
                          class="msg label"
                          data-tooltip="inherit"
                        >
                          Open Connection
                        </label>
                      </div>
                    </div>,
                  }
                }
                onDragOver={[Function]}
                onDrop={[Function]}
                orientation="row"
              >
                <div
                  className="msg container tabArea"
                  onDragOver={[Function]}
                  onDrop={[Function]}
                  style={
                    {
                      "alignItems": undefined,
                      "flexDirection": "row",
                      "flexWrap": undefined,
                      "justifyContent": undefined,
                    }
                  }
                >
                  <Button
                    className="tabItem selected"
                    draggable={true}
                    focusOnClick={false}
                    id="progress"
                    key="progress"
                    onClick={[Function]}
                    onDragEnter={[Function]}
                    onDrop={[Function]}
                    tabIndex={-1}
                  >
                    <div
                      className="msg button tabItem selected"
                      draggable={true}
                      id="progress"
                      onClick={[Function]}
                      onDragEnter={[Function]}
                      onDragLeave={[Function]}
                      onDragOver={[Function]}
                      onDragStart={[Function]}
                      onDrop={[Function]}
                      onMouseDown={[Function]}
                      role="button"
                      style={
                        {
                          "flexDirection": undefined,
                        }
                      }
                      tabIndex={-1}
                    >
                      <Icon
                        data-tooltip="inherit"
                        disabled={false}
                        src="overviewPage.svg"
                      >
                        <span
                          className="msg icon"
                          data-tooltip="inherit"
                          style={
                            {
                              "WebkitMaskImage": "url(overviewPage.svg)",
                              "backgroundColor": undefined,
                              "height": undefined,
                              "maskImage": "url(overviewPage.svg)",
                              "width": undefined,
                            }
                          }
                        />
                      </Icon>
                      <Label
                        data-tooltip="inherit"
                      >
                        <label
                          className="msg label"
                          data-tooltip="inherit"
                          style={{}}
                        >
                          Open Connection
                        </label>
                      </Label>
                    </div>
                  </Button>
                </div>
              </Container>
              <div
                className="scrollbar"
              >
                <div
                  className="slider"
                  onPointerDown={[Function]}
                  onPointerMove={[Function]}
                  onPointerUp={[Function]}
                />
              </div>
            </div>
          </div>
        </Container>
        <Container
          className="content"
          innerRef={
            {
              "current": <div
                class="msg container fixedScrollbar content"
                style="flex-direction: column;"
              >
                <div
                  class="msg container fixedScrollbar progressIndicatorHost"
                  id="loadingProgressIndicator"
                  style="flex-direction: column; justify-content: center; align-items: center; opacity: 0.95; --position: 0; --width: 80; --height: 80; --strokeWidth: 5;"
                >
                  <svg
                    class="circleBackground animated"
                  >
                    <circle
                      cx="37.5"
                      cy="37.5"
                      r="35"
                    />
                    <circle
                      cx="37.5"
                      cy="37.5"
                      r="35"
                    />
                  </svg>
                </div>
                <label
                  class="msg label"
                  data-tooltip="expand"
                  id="progressMessageId"
                />
              </div>,
            }
          }
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          orientation="column"
        >
          <div
            className="msg container fixedScrollbar content"
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            style={
              {
                "alignItems": undefined,
                "flexDirection": "column",
                "flexWrap": undefined,
                "justifyContent": undefined,
              }
            }
          >
            <ProgressIndicator
              backgroundOpacity={0.95}
              id="loadingProgressIndicator"
              linear={false}
            >
              <Container
                className="msg progressIndicatorHost"
                crossAlignment="center"
                id="loadingProgressIndicator"
                mainAlignment="center"
                orientation="column"
                style={
                  {
                    "--height": 80,
                    "--position": 0,
                    "--strokeWidth": 5,
                    "--width": 80,
                    "opacity": 0.95,
                  }
                }
              >
                <div
                  className="msg container fixedScrollbar progressIndicatorHost"
                  id="loadingProgressIndicator"
                  style={
                    {
                      "--height": 80,
                      "--position": 0,
                      "--strokeWidth": 5,
                      "--width": 80,
                      "alignItems": "center",
                      "flexDirection": "column",
                      "flexWrap": undefined,
                      "justifyContent": "center",
                      "opacity": 0.95,
                    }
                  }
                >
                  <svg
                    className="circleBackground animated"
                  >
                    <circle
                      cx={37.5}
                      cy={37.5}
                      r={35}
                    />
                    <circle
                      cx={37.5}
                      cy={37.5}
                      r={35}
                    />
                  </svg>
                </div>
              </Container>
            </ProgressIndicator>
            <Label
              caption=""
              id="progressMessageId"
            >
              <label
                className="msg label"
                data-tooltip="expand"
                id="progressMessageId"
                style={{}}
              />
            </Label>
          </div>
        </Container>
      </div>
    </Container>
  </Tabview>
</DBEditorModule>
`;
